GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs

```diff
@@ -549,7 +549,9 @@ public IReadOnlyList<ObjectId> GetRevisionChildren(ObjectId objectId)
             // With lock, loading the commit info slows down terribly.
             if (_revisionGraph.TryGetNode(objectId, out var node))
             {
-                return node.Children.Select(d => d.GitRevision.ObjectId).ToList();
+                var children = node.Children.Select(d => d.GitRevision.ObjectId).ToList();
+                children.Reverse();
```

Comment:
📝 The iteration order for `ImmutableStack<T>` is LIFO, which is the reverse of the collection used previously. This is not a problem; the other performance characteristics of `ImmutableStack<T>` ended up outweighing the benefits `ImmutableQueue<T>` would have provided to keep the same iteration order.

Reply:
Curious, would it not be faster to do:
```
var children = node.Children.Select(d => d.GitRevision.ObjectId).Reverse().ToList();
```
instead? Have the enumerator create the list in reverse in one go?

Reply:
Probably not. The implementation of `Reverse` would need to buffer everything in an intermediate collection that then gets discarded. Reversing the list itself is an in-place operation.